### DISCUSSION TITLE
More concrete & runnable examples for EVM package (like for trie)

### DISCUSSION THREAD

**Discussion by faustbrian** [ethereumjs/ethereumjs-monorepo#2125] [Q&A]
So I'm currently looking at https://github.com/ethereumjs/ethereumjs-monorepo/tree/master/packages/vm and https://github.com/ethereumjs/ethereumjs-monorepo/tree/master/packages/evm because I'm planning to do an integration of those into an application in the near future but it's all fairly dry and theoretical when you take a first look at those because there aren't really any hands on examples for executing example implementations of popular smart contracts like ERC-20, ERC-721, ERC-1155 to get a feel for what it takes to run those and to get a first taste of what is possible.

Is anything preventing these examples from being added or is that just something that never had any priority for some reason? Would provide a nicer introduction to those if you can actually get a feel for what is possible by cloning the repository and running something like an ERC-721 example to see the whole process run through and create something and at the same time see what's involved in making it happen. In general some more complete and runnable examples would be useful, for example for use-cases like using the EVM only to execute transactions and have it manage the state of accounts and transactions but never running blocks because that happens outside of the EVM context.

**Comment by holgerd77** [comment]
Thanks for bringing this up, yes, that's really only a priority issue, so it would be really great and super useful to have more practical examples here.
